corehq/motech/repeaters/tests/test_repeater.py

```diff
@@ -1284,6 +1287,65 @@ def test_five_retries(self):
             self.assertEqual(interval, timedelta(hours=expected_interval_hours))
 
 
+class DataSourceRepeaterTest(BaseRepeaterTest, TestXmlMixin):
+    domain = "user-reports"
+
+    def setUp(self):
+        super().setUp()
+        self.config = get_sample_data_source()
+        self.config.save()
+        self.adapter = get_indicator_adapter(self.config)
+        self.adapter.build_table()
+        self.fake_time_now = datetime(2015, 4, 24, 12, 30, 8, 24886)
+        self.pillow = _get_pillow([self.config], processor_chunk_size=100)
+
+        self.connx = ConnectionSettings.objects.create(
+            domain=self.domain,
+            url="case-repeater-url",
+        )
+        self.data_source_id = self.config._id
+        self.repeater = DataSourceRepeater(
+            domain=self.domain,
+            connection_settings_id=self.connx.id,
+            repeater_id=uuid.uuid4().hex,
+            data_source_id=self.data_source_id,
+        )
+        self.repeater.save()
+
+    def test_datasource_is_subscribed_to(self):
+        self.assertTrue(self.repeater.datasource_is_subscribed_to(self.domain, self.data_source_id))
+        self.assertFalse(self.repeater.datasource_is_subscribed_to("malicious-domain", self.data_source_id))
+
+    @flag_enabled('SUPERSET_ANALYTICS')
+    def test_payload_format(self):
+        doc_id = self._create_log_and_repeat_record()
+        later = datetime.utcnow() + timedelta(hours=50)
+        repeat_record = RepeatRecord.all(domain=self.domain, due_before=later).first()
+        payload = self.repeater.get_payload(repeat_record)
+        payload = json.loads(payload)
+        self.assertEqual(payload["data_source_id"], self.data_source_id)
+        self.assertTrue(isinstance(payload["data"], list))
```

Comment:
can we test for the actual data in it?

Reply:
2c7d316dd79acca4967526f08bedd20ca0887451